tokenscan/scanner.go

```diff
@@ -0,0 +1,221 @@
+package tokenscan
+
+import (
+	"context"
+	"math"
+	"math/rand"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/gocql/gocql"
+
+	"github.com/scylladb/gocqlx"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/scylladb/gocqlx/qb"
+)
+
+const (
+	// parallelism = nodesInCluster * coresInNode * 3
+	defaultParallelism = 72
+)
+
+type tokenRange struct {
+	startRange int64
+	endRange   int64
+}
+
+// SessionBuilder creates a gocql.Session
```

Comment:
Single session should be used, there is no point in opening more TCP connections from one app.